server/src/main/java/io/crate/metadata/cluster/AlterTableClusterStateExecutor.java

```diff
@@ -108,7 +108,7 @@ protected ClusterState execute(ClusterState currentState, AlterTableRequest requ
         if (request.isPartitioned()) {
             if (request.partitionIndexName() != null) {
                 assert request.mappingDelta() == null
-                    : "Cannot add column to a single partition. Template and index mappings must stay in sync";
+                    : "Cannot SET column policy to a single partition.";
```

Comment:
Even though this class will be used only for `ALTER TABLE SET`, changing column policy of the table still requires mapping update and we can have not null `request.mappingDelta()`.

However, we cannot set column policy to a single partition, see relevant [code](https://github.com/crate/crate/blob/master/server/src/main/java/io/crate/planner/node/ddl/AlterTablePlan.java#L119-L123). 

Scenario _"Setting column policy on a concrete partition is not possible"_ is actually already checked in analyzer which throws an error (invalid setting...) but since we had an assertion here I kept it in case this check will be somehow deleted in analyzer by accident.